app/src/main/java/com/example/drawingactivity/MainActivity.kt

```diff
@@ -99,4 +110,85 @@ class MainActivity : AppCompatActivity() {
         }
         dialog.show()
     }
+
+    private fun editLayerDialog(oldLayerName: String) {
```

Comment:
is this your add layer function as well? If it isn't your supposed to try to reuse the add layer to edit the item (both are inherently getting a string and setting it to some place)